projects/extension/ai/semantic_catalog.py

```diff
@@ -0,0 +1,383 @@
+import json
+from collections.abc import Generator
+from typing import TypedDict
+
+
+class GeneratedDescription(TypedDict):
+    name: str
+    description: str
+
+
+def render_obj_sample(plpy, relation: str) -> str:
+    result = plpy.execute(f"select * from {relation}", 5)
+
+    ret_obj = f"""<data id="{relation}">"""
+    for r in result:
+        ret_obj += f"""\n  insert into {relation} ({', '.join(r.keys())}) values ({', '.join([str(v) for v in r.values()])});"""
```

Comment:
Use the utility functions to quote the table name and quote the values as literals.

- plpy.quote_ident()
- plpy.quote_literal()
- plpy.quote_nullable()

https://www.postgresql.org/docs/current/plpython-util.html

Reply:
Reworked this in 4967a3d51918ff09a597e0f28a1e6daa6a6826a6. Though, I wonder if don't need to use `plpy.quote_ident` for `relation`, given that it's a fully quantified `schema_name.table_name` structure. If need to use it, then need to first split into two separate parts, and then apply `plpy.quote_ident` to each part.

I did the latter in 4967a3d51918ff09a597e0f28a1e6daa6a6826a6, happy to revise though.